java/src/shared/model/game/TurnTracker.java

```diff
@@ -61,11 +68,41 @@ public int getCurrentTurn() {
     }
 
     /**
+     * Goes to the next players turn during the setup phase.
+     * @return The index of the next player to setup
+     * @throws Exception if setupTurn > getNumPlayers()
+     */
+    public int nextSetupTurn() throws Exception {
+        if (status.equals("FirstRound")) {
+            if (setupTurn < getNumPlayers()) {
+                return setupTurn++;
+            } else if (setupTurn == numPlayers) {
+                status = "SecondRound";
+                return setupTurn;
+            } else {
+                throw new Exception("Current setup turn is invalid.  This is broken.");
+            }
+        } else if (status.equals("SecondRound")) {
+            return setupTurn--;
+        } else {
+            throw new Exception("Invalid status. Can only do setup when status is 'FirstRound' or 'SecondRound'");
+        }
+    }
+
```

Comment:
perhaps make canRoll, canTrade, canBuild which checks the if the status is the number (phase) we want.
Then an incrementStatus function